luthier/src/main/java/com/yahoo/luthier/webservice/data/config/dimension/DefaultDimensionTemplate.java

```diff
@@ -0,0 +1,151 @@
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+import com.yahoo.bard.webservice.util.EnumUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Dimension template.
+ * <p>
+ * An example:
+ * <p>
+ *      {
+ *          "apiName": "REGION_ISO_CODE",
+ *          "longName": "wiki regionIsoCode",
+ *          "description": "Iso Code of the region to which the wiki page belongs",
+ *          "fields": "default"
+ *      }
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DefaultDimensionTemplate implements DimensionTemplate {
+
+    private final String apiName;
+    private final String description;
+    private final String longName;
+    private final String category;
+    private DimensionFieldListTemplate fields;
+
+    /**
+     * Constructor used by json parser.
+     *
+     * @param apiName json property apiName
+     * @param description json property description
+     * @param longName json property longName
+     * @param category json property category
+     * @param fields json property fields deserialize by DimensionFieldDeserializer
+     */
+    @JsonCreator
+    public DefaultDimensionTemplate(
+            @NotNull @JsonProperty("apiName") String apiName,
+            @JsonProperty("description") String description,
+            @JsonProperty("longName") String longName,
+            @JsonProperty("category") String category,
+            @JsonProperty("fields") DimensionFieldListTemplate fields
+    ) {
+        this.apiName = EnumUtils.camelCase(apiName);
+        this.description = (Objects.isNull(description) ? "" : description);
+        this.longName = (Objects.isNull(longName) ? EnumUtils.camelCase(apiName) : longName);
+        this.category = (Objects.isNull(category) ? Dimension.DEFAULT_CATEGORY : category);
+        this.fields = fields;
+    }
+
+    /**
+     * Get dimensions info.
+     */
+    @Override
+    public String getApiName() {
+        return this.apiName;
+    }
+
+    @Override
+    public String getDescription() {
+        return this.description;
+    }
+
+    @Override
+    public String getLongName() {
+        return this.longName;
+    }
+
+    @Override
+    public String getCategory() {
+        return this.category;
+    }
+
+    @Override
+    public String toString() {
+        return this.getApiName();
+    }
+
+    @Override
+    public LinkedHashSet<DimensionField> getFields(
+            Map<String, LinkedHashSet<DimensionFieldInfoTemplate>> fieldDictionary
+    ) {
+        resolveFields(fieldDictionary);
+        if (this.fields.getFieldList() == null) {
+            return new LinkedHashSet<>();
+        }
+        return this.fields.getFieldList().stream()
+                .map(DimensionFieldInfoTemplate::build)
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    @Override
+    public DimensionName build() {
+        return this::getApiName;
+    }
```

Comment:
Ok, so I think this should NOT be returning a dimension name, but should instead be returning a `DimensionConfig`. 
For this default implementation you should probably be constructing a `DefaultKeyValueStoreDimensionConfig`. That being said, I am not sure how to handle the key value store nor the search provider that are passed into this.

@michael-mclawhorn any suggestions?